modules/references/tests/server/reference.server.model.tests.js

```diff
@@ -0,0 +1,181 @@
+'use strict';
+
+var should = require('should'),
+    mongoose = require('mongoose'),
+    User = mongoose.model('User'),
+    Reference = mongoose.model('Reference');
+
+describe('Reference Model Unit Tests', function () {
+
+  describe('Method Save', function () {
+
+    var user1,
+        user2,
+        user3;
+
+    beforeEach(function () {
+      user1 = new User({
+        username: 'user1',
+        email: 'user1@example.com',
+        password: 'correcthorsebatterystaples'
+      });
+
+      user2 = new User({
+        username: 'user2',
+        email: 'user2@example.com',
+        password: 'correcthorsebatterystaples'
+      });
+
+      user3 = new User({
+        username: 'user3',
+        email: 'user3@example.com',
+        password: 'correcthorsebatterystaples'
+      });
+    });
+
+    afterEach(function (done) {
+      Reference.remove().exec(done);
+    });
+
+    it('save both directions without problems', function (done) {
+      var reference1 = new Reference({
+        userFrom: user1._id,
+        userTo: user2._id,
+        interactions: {
+          met: true,
+          hostedMe: true,
+          hostedThem: false
+        },
+        recommend: 'no'
+      });
+
+      var reference2 = new Reference({
+        userFrom: user2._id,
+        userTo: user1._id,
+        interactions: {
+          met: true,
+          hostedMe: false,
+          hostedThem: true
+        },
+        recommend: 'yes'
+      });
+
+      reference1.save(function (err) {
+        try {
+          should.not.exist(err);
+        } catch (e) {
+          return done(e);
+        }
+        reference2.save(function (err) {
+          try {
+            should.not.exist(err);
+          } catch (e) {
+            return done(e);
+          }
+          return done();
+        });
+      });
+    });
+
+    it('save multiple references from one user to different users without problems', function (done) {
+      var reference1 = new Reference({
+        userFrom: user1._id,
+        userTo: user2._id,
+        interactions: {
+          met: true,
+          hostedMe: true,
+          hostedThem: false
+        },
+        recommend: 'no'
+      });
+
+      var reference2 = new Reference({
+        userFrom: user1._id,
+        userTo: user3._id,
+        interactions: {
+          met: true,
+          hostedMe: false,
+          hostedThem: true
+        },
+        recommend: 'yes'
+      });
+
+      reference1.save(function (err) {
+        try {
+          should.not.exist(err);
+        } catch (e) {
+          return done(e);
+        }
+        reference2.save(function (err) {
+          try {
+            should.not.exist(err);
+          } catch (e) {
+            return done(e);
+          }
```

Comment:
I don't think we have try/catch in our other assertion tests. Not opposed to this but I'm curious to hear rationale here? Somehow I remember earlier discussions about this but it escapes me now...

Reply:
If the assertion fails, it will lead to test timeout (uncaught error in asynchronous code). Indeed it's missing elsewhere, but I need this when practicing test first development. Can't always wait for the timeout. :)

This is a theory, I'll check it.

Reply:
Ok, it's actually caught and no timeout happens. Magic. I can simplify them, then.

:+1:

Reply:
> Nice! I was trying to understand this better from Mocha docs but didn't see need for this there neither.

The only difference is that the report says: `Uncaught Error` instead of `Error`.

With Promises/(async functions), this stuff will be a breeze. :slightly_smiling_face:

Reply:
I'm confused, because now I encountered a case which timeouts without try catch.

Remove `try catch` [here](https://github.com/Trustroots/trustroots/pull/723/files#diff-6af870a19c75bac875ded8cb33478262R208) and change `length(1)` to `length(0)`.

I'm tempted to leave this as it is, because it doesn't cause any harm, keeps things predictable and is likely to be refactored as part of #804.

Reply:
Sounds good! :thumbsup:

Reply:
Nah, Github email reply thew my reply out of the thread: https://github.com/Trustroots/trustroots/pull/723#issuecomment-435608095

Reply:
I'm marking this as resolved, then. :slightly_smiling_face: